packages/tools/pactjs-cli/src/contract-generate/tests/generate.test.ts

```diff
@@ -0,0 +1,200 @@
+jest.mock('fs');
+jest.mock('cross-fetch');
+jest.mock('../../utils/retrieveContractFromChain');
+
+import { retrieveContractFromChain } from '../../utils/retrieveContractFromChain';
+import { generate } from '../generate';
+import { ContractGenerateOptions } from '..';
+
+import { mockContract } from './mockdata/contract';
+
+import { Command, program } from 'commander';
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+
+const mockedReadFileSync = readFileSync as jest.MockedFunction<
+  typeof readFileSync
+>;
+mockedReadFileSync.mockImplementation(
+  mockedReadFileSync as jest.MockedFunction<typeof readFileSync>,
+);
+
+const mockedWriteFileSync = writeFileSync as jest.MockedFunction<
+  typeof writeFileSync
+>;
+mockedWriteFileSync.mockImplementation(
+  mockedWriteFileSync as jest.MockedFunction<typeof writeFileSync>,
+);
+
+const mockedExistsSync = existsSync as jest.MockedFunction<typeof existsSync>;
+mockedExistsSync.mockImplementation(
+  mockedExistsSync as jest.MockedFunction<typeof existsSync>,
+);
+
+const mockedRetrieveContractFromChain =
+  retrieveContractFromChain as jest.MockedFunction<
+    typeof retrieveContractFromChain
+  >;
+mockedRetrieveContractFromChain.mockImplementation(
+  mockedRetrieveContractFromChain as jest.MockedFunction<
+    typeof retrieveContractFromChain
+  >,
+);
```

Comment:
Maybe something like `mock-fs` could be used? https://www.npmjs.com/package/mock-fs

Reply:
I'd like to suggest keeping these these changes out of this PR. Streamlining the tests should be much easier after #197 or similar.